third_party/move/move-prover/src/lib.rs

```diff
@@ -71,19 +71,26 @@ pub fn run_move_prover_v2<W: WriteColor>(
     options: Options,
 ) -> anyhow::Result<()> {
     let now = Instant::now();
-    let cloned_options = options.clone();
+    let mut env = create_move_prover_v2_model(error_writer, options.clone())?;
+    run_move_prover_with_model_v2(&mut env, error_writer, options, now)
+}
+
+pub fn create_move_prover_v2_model<W: WriteColor>(
+    error_writer: &mut W,
+    options: Options,
+) -> anyhow::Result<GlobalEnv> {
     let compiler_options = move_compiler_v2::Options {
-        dependencies: cloned_options.move_deps,
-        named_address_mapping: cloned_options.move_named_address_values,
-        output_dir: cloned_options.output_path,
-        language_version: cloned_options.language_version,
+        dependencies: options.move_deps,
+        named_address_mapping: options.move_named_address_values,
+        output_dir: options.output_path,
+        language_version: options.language_version,
         compiler_version: None, // TODO: need to pass v2.x here
```

Comment:
Just caught might eye (even if this is not related to this PR), we might want to use latest_stable compiler version here (or remove the TODO). Right now, it is probably being set based on the language version.